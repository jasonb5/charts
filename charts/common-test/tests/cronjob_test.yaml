suite: test cronjob
values:
  - values/common.yaml
templates:
  - app.yaml
chart:
  appVersion: 0.1.0
tests:
  - it: should overwrite defaults
    set:
      workload: cronjob

      concurrencyPolicy: Allow
      failedJobHistoryLimit: 10
      schedule: "* * * * *"
      startingDeadlineSeconds: 10
      successfulJobHistoryLimit: 10
      timeZone: America/Los_Angeles
    asserts:
      - matchSnapshot:
          path: spec
        documentIndex: 0
      - hasDocuments:
          count: 2
  - it: should pass
    set:
      workload: cronjob
    asserts:
      - matchSnapshot: {}
        documentIndex: 0
      - hasDocuments:
          count: 2
