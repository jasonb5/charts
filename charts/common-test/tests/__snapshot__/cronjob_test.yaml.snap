should overwrite defaults:
  1: |
    concurrencyPolicy: Allow
    failedJobHistoryLimit: 10
    jobTemplate:
      spec:
        template:
          spec:
            containers:
              - image: busybox:0.1.0
                name: common-test
    schedule: '* * * * *'
    startingDeadlineSeconds: 10
    successfulJobHistoryLimit: 10
    timeZone: America/Los_Angeles
should pass:
  1: |
    apiVersion: batch/v1
    kind: CronJob
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      jobTemplate:
        spec:
          template:
            spec:
              containers:
                - image: busybox:0.1.0
                  name: common-test
      schedule: null
